### DISCUSSION TITLE
Making `asynq` working locally without Redis

### DISCUSSION THREAD

**Discussion by davidroman0O** [hibiken/asynq#731] [Show and tell]
Hi

I love this repository and library, it's super handy to build queues systems! Therefore, I really wanted to be able to use it on any kind of app even without Redis itself. Why bothering re-implementing a job system or use sqlite with `cgo`?! 

I'm working on a side project leveraging that idea but that's not the goal this post.

I just wanted to share that I've been using [miniredis](https://github.com/alicebob/miniredis) with `asynq` to build an app and it is working now.

The main issue was `asynq dash` and most calls like `GetQueueInfo` will trigger an error due to a missing call to `MEMORY USAGE` command on [miniredis](https://github.com/alicebob/miniredis) but [a PullRequest was done ](https://github.com/alicebob/miniredis/pull/341) to implement a somewhat similar kind of call.

I'm happy to say that you can now perfectly use `asynq dash` to debug your queue on a codebase that use `asynq` and `miniredis` at the same time to build single binary application and still reuse a bunch of code if one day you need to add a real instance of redis.

<img width="978" alt="working" src="https://github.com/hibiken/asynq/assets/2845356/c97f012f-9e02-4a52-912c-1b416c4cb57e">

That's it! Just wanted to notify that the pull request made it work :)  

Thank you @hibiken for your work, I'm glad to sponsor you to continue maintaining `asynq`

**Comment by hibiken** [comment]
@davidroman0O this is awesome, thank you for sharing this news!
I think some people may have similar use-case and be interested to know more about this.
Please feel free to add content in README and/or Wiki page!
